sc-memory/sc-core/src/sc-store/sc-event/sc_event_queue.c

```diff
@@ -95,6 +96,31 @@ void _sc_event_emission_pool_worker(sc_pointer data, sc_pointer user_data)
 
   sc_storage_end_new_process();
 
+  if (SC_ADDR_IS_EQUAL(event_subscription->event_type_addr, sc_event_before_erase_connector_addr)
+      || SC_ADDR_IS_EQUAL(event_subscription->event_type_addr, sc_event_before_erase_incoming_arc_addr)
+      || SC_ADDR_IS_EQUAL(event_subscription->event_type_addr, sc_event_before_erase_outgoing_arc_addr)
+      || SC_ADDR_IS_EQUAL(event_subscription->event_type_addr, sc_event_before_erase_edge_addr)
+      || SC_ADDR_IS_EQUAL(event_subscription->event_type_addr, sc_event_before_erase_element_addr))
+  {
+    sc_monitor_acquire_write(&queue->pool_monitor);
+    sc_uint32 * count = (sc_uint32 *)sc_hash_table_get(
+        queue->emitted_erase_events, GUINT_TO_POINTER(SC_ADDR_LOCAL_TO_INT(event->connector_addr)));
+    if (count != null_ptr)
+    {
+      --(*count);
+      if (*count == 0)
+      {
+        sc_hash_table_remove(
+            queue->emitted_erase_events, GUINT_TO_POINTER(SC_ADDR_LOCAL_TO_INT(event->connector_addr)));
+        sc_mem_free(count);
+      }
+      else
+        sc_hash_table_insert(
+            queue->emitted_erase_events, GUINT_TO_POINTER(SC_ADDR_LOCAL_TO_INT(event->connector_addr)), count);
```

Comment:
I don't sure that is correct name for this variable

Reply:
active_erase_event_subscription_callbacks_count
and it will behave unexpectedly for sc_event_before_erase_element because connector_addr is used as key in this map(which is SC_ADDR_EMPTY in that case), I think that if connector_addr is SC_ADDR_EMPTY then event->event_addr can be used as key because that element is being erased, what do you think about that?

Reply:
okay, agree with you